src/nnet3/nnet-simple-component.h

```diff
@@ -157,6 +161,9 @@ class NormalizeComponent: public NonlinearComponent {
   // about 0.7e-20.  We need a value that's exactly representable in
   // float and whose inverse square root is also exactly representable
   // in float (hence, an even power of two).
+
+  bool add_log_std_; // If true, log(max(epsi, sqrt(row_in^T row_in / D)))  
```

Comment:
you should call it add_log_stddev_, not add_log_std_.
Also, " is added as node to output layer." is not the right terminology as used in nnet3, its an extra dimension in the output layer.